MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -262,6 +278,35 @@ public DepthStencilState DepthStencilState
             }
         }
 
+        internal void ApplyState(bool applyShaders)
+        {
+            PlatformBeginApplyState();
+
+            if (_blendStateDirty)
+            {
+                // Static state properties never actually get bound;
+                // instead we use our GraphicsDevice-specific version of them.
+                var newBlendState = _blendState;
+                if (ReferenceEquals(_blendState, BlendState.Additive))
+                    newBlendState = BlendStateAdditive;
+                else if (ReferenceEquals(_blendState, BlendState.AlphaBlend))
+                    newBlendState = BlendStateAlphaBlend;
+                else if (ReferenceEquals(_blendState, BlendState.NonPremultiplied))
+                    newBlendState = BlendStateNonPremultiplied;
+                else if (ReferenceEquals(_blendState, BlendState.Opaque))
+                    newBlendState = BlendStateOpaque;
+
+                // Blend state is now bound to a device... no one should
+                // be changing the state of the blend state object now!
+                newBlendState.BindToGraphicsDevice(this);
+
+                newBlendState.PlatformApplyState(this);

```

Comment:
What is the benefit of separating the `BindToGraphicsDevice` and `PlatformApplyState`?